schema/ext/vulnerability-1.0.xsd

```diff
@@ -0,0 +1,148 @@
+        </xs:sequence>
+    </xs:complexType>
+
+    <xs:complexType name="cvssV3">
+        <xs:annotation>
+            <xs:documentation>
+                Defines the structure of a CVSS v3 vulnerability
+                https://www.first.org/cvss/v3.1/specification-document
+            </xs:documentation>
+        </xs:annotation>
+        <xs:sequence>
+            <xs:element name="baseScore" type="xs:decimal" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="attackVector" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="attackComplexity" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="privilegesRequired" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="userInteraction" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="scope" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="confidentialityImpact" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="integrityImpact" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="availabilityImpact" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="baseSeverity" type="xs:string" minOccurs="1" maxOccurs="1"/>
+        </xs:sequence>
+    </xs:complexType>
+
+    <xs:complexType name="vulnerability">
+        <xs:annotation>
+            <xs:documentation>Defines the structure of a vulnerability</xs:documentation>
+        </xs:annotation>
+        <xs:sequence>
+            <xs:element name="id" type="xs:normalizedString" minOccurs="1" maxOccurs="1">
+                <xs:annotation>
+                    <xs:documentation>The id of the vulnerability</xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="source" minOccurs="0" maxOccurs="1">
+                <xs:complexType>
+                    <xs:annotation>
+                        <xs:documentation>The source of the vulnerability</xs:documentation>
+                    </xs:annotation>
+                    <xs:sequence>
+                        <xs:element name="url" type="xs:string" minOccurs="0" maxOccurs="1">
+                            <xs:annotation>
+                                <xs:documentation>The url of the source</xs:documentation>
+                            </xs:annotation>
+                        </xs:element>
+                    </xs:sequence>
+                    <xs:attribute name="name" type="xs:normalizedString" use="required">
+                        <xs:annotation>
+                            <xs:documentation>The name of the source</xs:documentation>
+                        </xs:annotation>
+                    </xs:attribute>
+                </xs:complexType>
+            </xs:element>
+            <xs:element name="cvssV2" type="dg:cvssV2" minOccurs="0" maxOccurs="1"/>
```

Comment:
Not all public vulnerabilities use CVSS. Some (e.g. NPM Advisories) use strings (Critical, High, Moderate, Low) to represent severity. Organizations that use BOMs within their environment typically will not use CVSS, rather OWASP Risk Rating or a similar methodology.

Reply:
> Not all public vulnerabilities use CVSS. Some (e.g. NPM Advisories) use strings (Critical, High, Moderate, Low) to represent severity. Organizations that use BOMs within their environment typically will not use CVSS, rather OWASP Risk Rating or a similar methodology.

👍 restructured the scoreType to be generic that can support multiple risk rating methodologies.